### PR TITLE
chore(upgrade): upgrade CKEditor to latest version 4.6.2

### LINKED ISSUE #9283: Update CKEditor & add "Code Snippet" addon
Hi!

I propose to update the "CKEditor" to the latest version.
And add "Code Snippet" addon: http://ckeditor.com/addon/codesnippet

I already did and not found problems.

### PR DESCRIPTION
**Pull request by smellems**
Closes #9283

### CODE REVIEW

**Comment by hypeJunction** [comment]
It's a breaking change and would need to go into master

**Comment by smellems** [comment]
What does it break?  I'm also looking at upgrading it in our 1.12..

**Comment by hypeJunction** [comment]
You are bringing a feature release of CKEditor in a bugfix branch of Elgg.

**Comment by mrclay** [comment]
In master branch we might as well drop all these vendor files. #10968

**Comment by mrclay** [comment]
> I'm also looking at upgrading it in our 1.12

You'll need to make a separate plugin anyway.

**Comment by smellems** [comment]
I changed this to update to 4.4.8 for branch 2.3.  I can make another pull request for 4.6.2, for master.

File: mod/ckeditor/views/default/elgg/ckeditor/config.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ define(function(require) {
 		removeButtons: 'Subscript,Superscript', // To have Underline back
 		allowedContent: true,
 		baseHref: elgg.get_site_url(),
-		removePlugins: 'liststyle,contextmenu,tabletools',
-		extraPlugins: 'blockimagepaste,autogrow',
```
Comment (mrclay) [inline mod/ckeditor/views/default/elgg/ckeditor/config.js]: Why are these removals being made? At the very least they should be mentioned in the PR description for transparency.

**Review by mrclay — CHANGES_REQUESTED** [review]

File: mod/ckeditor/views/default/elgg/ckeditor/config.js (hunk 1 above)
Comment (smellems) [inline mod/ckeditor/views/default/elgg/ckeditor/config.js]: `blockimagepaste `is not a plugin in the plugin folder, same for `contextmenu`.  `liststyle `and `tabletools `were already being removed.

#9172

Comment (mrclay) [inline mod/ckeditor/views/default/elgg/ckeditor/config.js]: I'm OK with this, but @Elgg/core should probably determine if the functionality these plugins were providing is still needed.

**Review by mrclay — APPROVED** [review]

**Comment by iionly** [comment]
May I suggest to include the "full package" of the CKEditor plus at least the plugins already in use by Elgg additionally (e.g. autogrow isn't part of the "full package") even if not all of these plugins are being used in the default Elgg ckeditor config.

I think it would be easier that way for anyone who wants to customize their editor config as it would only be necessary to override the views and not to create a complete plugin that contains the plugin missing in the bundled ckeditor plugin. With https://github.com/hypeJunction/Elgg-ckeditor_addons there's even a plugin already available that allows to define the editor config via the admin backend and such a plugin would fail to work if some of the required plugins wouldn't be included while the corresponding features could still be selected.

**Comment by hypeJunction** [comment]
This should go into master in any case, I don't think there is place for these changes in a bugfix release. And while at it, as @mrclay suggested, vendor libraries should be pulled via bower or something. Perhaps it would make sense to extract the plugin into its own repo and pull via composer.

**Comment by smellems** [comment]
This is the "full package" with the `autogrow` plugin addded.

I can also make the change in master.  Not sure how you would like it.

**Comment by jdalsem** [comment]
Replaced by #11453

**Pull request closed without merging**